sympy/simplify/simplify.py

```diff
@@ -654,43 +677,89 @@ def sum_simplify(s):
                 for j, s_term2 in enumerate(s_t):
                     if not used[j] and i != j:
                         temp = sum_add(s_term1, s_term2, method)
-                        if isinstance(temp, Sum):
+                        if isinstance(temp, Sum) or isinstance(temp, Mul):
                             s_t[i] = temp
                             s_term1 = s_t[i]
                             used[j] = True
 
-    result = Add(*o_t)
+    return used, s_t
 
-    for i, s_term in enumerate(s_t):
-        if not used[i]:
-            result = Add(result, s_term)
+def sum_factor(self, limits=None):
```

Comment:
Yes. Instead of "factor" (which for SymPy means factoring a sum into a product), it should be "expand".

For expand, it means calling this from one of the `Sum._eval_expand` methods.